app/src/main/cpp/skyline/services/base_service.h

```diff
@@ -138,39 +28,48 @@ namespace skyline::service {
       protected:
         const DeviceState &state; //!< The state of the device
         ServiceManager &manager; //!< A reference to the service manager
-        std::unordered_map<u32, std::function<void(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> vTable; //!< This holds the mapping from a function's CmdId to the actual function
+        std::unordered_map<u32, std::function<Result(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> vTable; //!< This holds the mapping from a function's CmdId to the actual function
 
       public:
-        Service serviceType; //!< The type of this service
-        std::string serviceName; //!< The name of this service
-
         /**
          * @param state The state of the device
-         * @param hasLoop If the service has a loop or not
-         * @param serviceType The type of the service
-         * @param serviceName The name of the service
          * @param vTable The functions of the service
          */
-        BaseService(const DeviceState &state, ServiceManager &manager, Service serviceType, const std::string &serviceName, const std::unordered_map<u32, std::function<void(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> &vTable) : state(state), manager(manager), serviceType(serviceType), serviceName(serviceName), vTable(vTable) {}
+        BaseService(const DeviceState &state, ServiceManager &manager, const std::unordered_map<u32, std::function<Result(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> &vTable) : state(state), manager(manager), vTable(vTable) {}
+
+        /**
+         * @brief This is used to enable RTTI for use with service related logging
```

Comment:
`... for use with service related logging` I think `... to be able to extract the name of the underlying class and ensure correct destruction order` is better